app/src/main/java/controlers/RestServices.java

```diff
@@ -0,0 +1,80 @@
+package controlers;
+
+import android.util.Base64;
+import android.widget.Toast;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import android.content.Context;
+
+import dto.Route;
+import dto.Stop;
+import dto.Departure;
+
+public class RestServices {
+    public JSONObject connectToRestService(String url_service, String param, Context context)
+    {
+        try {
+            if(url_service == null || url_service.length() == 0)
+                throw new Exception("URL invalid");
+
+            URL url = new URL(url_service);
+            HttpURLConnection connectionHttp = (HttpURLConnection) url.openConnection();
+            connectionHttp.setRequestMethod("POST");
+            connectionHttp.setRequestProperty("X-AppGlu-Environment", "staging");
+            connectionHttp.setRequestProperty("Content-Type", "application/json");
+            String base64EncodedCredentials = Base64.encodeToString(
+                    ("WKD4N7YMA1uiM8V:DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68").getBytes(), Base64.DEFAULT);
```

Comment:
Is this username and password?

It's totally unreadable. In this cases you should extract a variable.
For example, if another guy get this code to fix a bug, he will never know what is it.